### PR TITLE
finished, but curious about how to make variables truly private

### PR DESCRIPTION
**Pull request by danieltichiyama**

### CODE REVIEW

File: zombies.js [hunk 1]
```diff
@@ -25,13 +28,17 @@
  * @property {number} damage
  */
 
+function Weapon(name, damage) {
+  this.damage = damage;
+  Item.call(this, name);
```
Comment (taesup) [inline zombies.js]: The `item.call` should be above the `this.damage`

File: zombies.js [hunk 2]
```diff
@@ -49,13 +56,17 @@
  * @property {number} energy
  */
 
+function Food(name, energy) {
+  this.energy = energy;
+  Item.call(this, name);
```
Comment (taesup) [inline zombies.js]: Move the Item.call one line up.

File: zombies.js [hunk 3]
```diff
@@ -79,6 +90,24 @@
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
 
+function Player(name, health, strength, speed) {
+  this._pack = [];
```
Comment (taesup) [inline zombies.js]: `_pack and _maxHealth` should be private variables by using const instead of this.

File: zombies.js [hunk 4]
```diff
@@ -159,6 +219,35 @@
...
+    this.getPack()[this.getPack().indexOf(itemToEquip)] = this.equipped;
+    console.log(
+      itemToEquip.name +
+        " is now equipped, and " +
+        this.equipped.name +
+        " has been stored."
+    );
+    this.equipped = itemToEquip;
+    return true;
+  }
```
Comment (taesup) [inline zombies.js]: Same comment about the last two if statements begin joined together.

File: zombies.js [hunk 5]
```diff
@@ -138,6 +186,18 @@
  * @return {boolean} true/false     Whether player was able to remove item from pack.
  */
 
+Player.prototype.discardItem = function(item) {
+  if (this.getPack().indexOf(item) !== -1) {
```
Comment (taesup) [inline zombies.js]: Same comment about putting the index into a variable here.

**Review by taesup — CHANGES_REQUESTED** [review]

File: zombies.js [hunk 6]
```diff
@@ -225,6 +350,13 @@
  * @property {boolean} isAlive      Default value should be `true`.
  */
 
+function Zombie(health, strength, speed) {
+  this._maxHealth = health;
```
Comment (taesup) [inline zombies.js]: This should be a const.

**Review by taesup — CHANGES_REQUESTED** [review]

File: zombies.js (hunk 6 above)
Comment (danieltichiyama) [reply zombies.js]: changes made, ready for review.

**Review by taesup — APPROVED** [review]

**Pull request merged**